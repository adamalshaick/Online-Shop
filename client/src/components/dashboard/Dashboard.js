import React from "react";
import PropTypes from "prop-types";
import { ButtonLink } from "../common/styles/Button";
import fetchCurrentUser from "../common/hoc/fetchCurrentUser";
import ProfileContent from "../profile/ProfileContent";
import Reviews from "../reviews/Reviews";
import { SecondaryHeader } from "../common/styles/Header";

export const Dashboard = ({ user }) => {
  return (
    <div className="container entry text-center">
      <div className="row mt-5">
        <div className="col-md-6">
          <SecondaryHeader>
            Your Profile <hr />
          </SecondaryHeader>
          <ProfileContent user={user.currentUser} />
          <SecondaryHeader className="mt-3">
            Actions <hr />
          </SecondaryHeader>
          <div className="mt-3">
            <ButtonLink to="/sell-item" className="btn btn-light btn-lg">
              Add your item for sale
            </ButtonLink>
            <span className="text-muted">Sell your item</span>
            <hr />
            <ButtonLink to="/items" className="btn btn-light btn-lg">
              Browse items
            </ButtonLink>
            <span className="text-muted">Browse items for sale</span>
            <hr />
          </div>
        </div>
        <div className="col-md-6">
          <SecondaryHeader>Reviews</SecondaryHeader> <hr />
          <Reviews user={user.currentUser} />
        </div>
      </div>
    </div>
  );
};

Dashboard.propTypes = {
  user: PropTypes.object.isRequired
};

export default fetchCurrentUser(Dashboard);
