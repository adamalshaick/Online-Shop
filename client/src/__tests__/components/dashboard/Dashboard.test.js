import React from "react";
import { Dashboard } from "../../../components/dashboard/Dashboard";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const Props = {
  user: {
    currentUser: {
      cart: []
    }
  }
};

const wrapper = shallow(<Dashboard {...Props} />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
