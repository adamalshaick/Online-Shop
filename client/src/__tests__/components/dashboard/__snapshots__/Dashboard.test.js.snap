// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="container entry text-center"
>
  <div
    className="row mt-5"
  >
    <div
      className="col-md-6"
    >
      <styled.header>
        Your Profile 
        <hr />
      </styled.header>
      <Connect(ProfileContent)
        user={
          Object {
            "cart": Array [],
          }
        }
      />
      <styled.header
        className="mt-3"
      >
        Actions 
        <hr />
      </styled.header>
      <div
        className="mt-3"
      >
        <Styled(Link)
          className="btn btn-light btn-lg"
          to="/sell-item"
        >
          Add your item for sale
        </Styled(Link)>
        <span
          className="text-muted"
        >
          Sell your item
        </span>
        <hr />
        <Styled(Link)
          className="btn btn-light btn-lg"
          to="/items"
        >
          Browse items
        </Styled(Link)>
        <span
          className="text-muted"
        >
          Browse items for sale
        </span>
        <hr />
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <styled.header>
        Reviews
      </styled.header>
       
      <hr />
      <Connect(ComposedComponent)
        user={
          Object {
            "cart": Array [],
          }
        }
      />
    </div>
  </div>
</div>
`;
